homeassistant/components/pvpc_hourly_pricing/__init__.py

```diff
@@ -0,0 +1,67 @@
+"""The pvpc_hourly_pricing integration to collect Spain official electric prices."""
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_NAME, CONF_TIMEOUT
+from homeassistant.core import HomeAssistant
+import homeassistant.helpers.config_validation as cv
+
+from .const import (
+    ATTR_TARIFF,
+    DEFAULT_NAME,
+    DEFAULT_TARIFF,
+    DEFAULT_TIMEOUT,
+    DOMAIN,
+    PLATFORM,
+    TARIFFS,
+)
+
+UI_CONFIG_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_NAME, default=DEFAULT_NAME): str,
+        vol.Required(ATTR_TARIFF, default=DEFAULT_TARIFF): vol.In(TARIFFS),
+    }
+)
+SINGLE_SENSOR_SCHEMA = UI_CONFIG_SCHEMA.extend(
+    {vol.Optional(CONF_TIMEOUT, default=DEFAULT_TIMEOUT): cv.positive_int}
```

Comment:
As developers we should pick a reasonable timeout for the user as they shouldn't have to know about the details in order to configure this.

Reply:
It is already picked, I was using this component for almost 2 years now, so it is already fine-tuned. 
Also, the `timeout` customization is only available when using manual yaml config.

But anyway, I'll remove that possibility too, and change to a fixed, conservative, timeout of 10s.